src/utils/openaiChat.ts

```diff
@@ -107,128 +110,144 @@ async function SummaryGenerate(question: string, ans: string): Promise<string> {
+                text = jsonData?.choices[0]?.delta?.content || ''
+                
+                const queue = encoder.encode(text);
+                controller.enqueue(queue);
+              } catch (e) {
+                controller.close()
+              }
+            }
+          }
+          const parser = createParser(onParse);
+          // https://web.dev/streams/#asynchronous-iteration
+          for await (const chunk of response.body as any) {
+            parser.feed(decoder.decode(chunk));
+          }
+        }
+      })
+
+      return stream
+
     } catch (error) {
       if (retry < 2) {
-        return SummaryGenerateCall(question, ans, retry + 1);
+        return SummaryGenerate(question, ans, link, retry + 1);
       } else {
-        return ERROR_MESSAGES.OVERLOAD;
+        return createReadableStream(ERROR_MESSAGES.OVERLOAD);
       }
     }
   }
 
-  return SummaryGenerateCall(question, ans);
-}
-
-function removeDuplicatesByID(arr: (CustomContent | null)[]): (CustomContent | null)[] {
+function removeDuplicatesByID(arr: CustomContent[]): CustomContent[] {
   const seen = new Set();
   const filteredArr = arr.filter((item) => {
-    if (item === null) return false;
     const isDuplicate = seen.has(item.link);
-    seen.add(item.link);
+    if (!isDuplicate) {
+      seen.add(item.link);
+    }
     return !isDuplicate;
   });
   return filteredArr;
 }
 
-async function getFinalAnswer(
-  question: string,
+function getFinalAnswer(
   summary: string,
   content: SummaryData[]
-): Promise<{ question: string; data: string }> {
+) {
   let data = summary.trim() + "\n\n";
 
   content.forEach((d: SummaryData, i: number) => {
     data += `[${i}]: ${d.link}\n`;
   });
 
-  return { question: question, data };
+  return { data };
 }
 
 export async function processInput(
-  searchResults: any[] | undefined,
+  searchResults: Result[] | undefined,
   question: string
-): Promise<string> {
+) {
   try {
-    if (!searchResults) {
+    if (!searchResults?.length) {
       let output_string: string = ERROR_MESSAGES.NO_ANSWER;
-      return output_string;
+      return createReadableStream(output_string);
     } else {
-      const intermediateContent: (CustomContent | null)[] = searchResults
-      .map((result: Result) => {
-        let results = result._source
+      const intermediateContent: CustomContent[] = []
+
+      for (const result of searchResults) {
+        let { _source: source} = result
         const isQuestionOnStackExchange =
-        results.type === "question" &&
-        results.url.includes("stackexchange");
-        const isMarkdown = results.body_type === "markdown";
-        const snippet = isMarkdown
-        ? concatenateTextFields(results.body)
-        : results.body;
-        return isQuestionOnStackExchange
-        ? null
-        : {
-          title: results.title,
-          snippet: snippet,
-          link: results.url,
-        };
-      });
+        source.type === "question" &&
```

Comment:
Can we add one more processing for the mailing-list? In this case, if it is a mailing-list then we use a summary of it as the final text from this result else we use the body if the summary is not present.

Reply:
Oh, okay. After the markdown check?

Reply:
yes